src/main/java/com/google/sps/data/AvailabilityTimeSlots.java

```diff
@@ -0,0 +1,138 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.TextStyle;
+import java.util.Locale;
+import java.time.format.DateTimeFormatter;
+
+// TODO: Add a test for the constructor.
+
+/**
+ * A collection of AvailabilityTimeSlot Objects, which will eventually be generated with information
+ * from datastore.
+ */
+public class AvailabilityTimeSlots {
+  private List<AvailabilityTimeSlot> timeSlots = new ArrayList<AvailabilityTimeSlot>();
+
+  /** Constructs an AvailabilityTimeSlots object by generating the timeSlots list. */
+  public AvailabilityTimeSlots(String timeZoneOffsetString) {
```

Comment:
This constructor is doing too much work.

http://go/java-practices/constructors#avoid-doing-work-in-constructors

> The simplest and commonest kind of constructor just assigns each of its parameters to a field of the same name and type, perhaps after validation. If your constructor does not look like that, consider whether it could. The more additional work the constructor does, the harder it will be to test the class, and especially to test other classes that need it. While mocking frameworks like Mockito allow you to get an instance of a class without calling its constructor, that requires the class not to be final, and can impose further constraints. Having a simple constructor gives test writers the option of constructing a real instance of the class without having to worry about whether the constructor will behave as expected.
> 
> Calling instance methods in a constructor also can introduce a certain class of problems where the new object can be observed in a partially-constructed state.
> 
> Instead of doing work in the constructor, consider whether it could be done in a factory method or class, in a builder, or when an `@Inject` constructor's parameters are derived by Guice or Dagger.
> 
> This is discussed at greater length in the CL Reviewer's Field Guide and in DI Practices.


(This is probably the root cause of why I want you to rename this class.)

Reply:
Class has been renamed to a Generator